### ISSUE TITLE
Different results returned when checking just CF versus both CF and ACDD

### ISSUE THREAD

**Issue by dnowacki-usgs** [ioos/compliance-checker#826]
With the Python interface, I use the following to test for CF-1.6 compliance:
```python
from compliance_checker.runner import ComplianceChecker, CheckSuite
check_suite = CheckSuite()
check_suite.load_all_available_checkers()
path = fildir + 'clean/' + filnam
# checker_names = ['acdd', 'cf:1.6']
checker_names = ['cf:1.6']
return_value, errors = ComplianceChecker.run_checker(path, checker_names, 0, 'normal')
```
and get this result:
```
Using cached standard name table v66 from /Users/dnowacki/.local/share/compliance-checker/cf-standard-name-table-test-66.xml
WARNING: The following exceptions occured during the cf:1.6 checker (possibly indicate compliance checker issues):
cf:1.6.check_data_types: type object 'str' has no attribute 'kind'


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
All tests passed!
```
There's a warning but all tests pass.

But if I change `checker_names` to `checker_names = ['acdd', 'cf:1.6']`, I get the following result:
```
Using cached standard name table v66 from /Users/dnowacki/.local/share/compliance-checker/cf-standard-name-table-test-66.xml
WARNING: The following exceptions occured during the cf:1.6 checker (possibly indicate compliance checker issues):
cf:1.6.check_data_types: type object 'str' has no attribute 'kind'


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 5 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "SED_981" missing the following attributes:
* coverage_content_type

variable "Trb_980" missing the following attributes:
* coverage_content_type

variable "z" missing the following attributes:
* long_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_email not present
* geospatial_bounds not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* license not present
* processing_level not present
* publisher_name not present
* publisher_email not present


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 1 potential issue


                                     Errors                                     
--------------------------------------------------------------------------------
§9.1 Features and feature types
* Unidentifiable feature for variable longitude
* Unidentifiable feature for variable latitude
* Unidentifiable feature for variable z
```
A bunch of ACDD actions, which is fine, but note that **the CF compliance output has changed and now has three errors**. What's going on? [The file is available here](https://geoport.usgs.esipfed.org/thredds//sand/usgs/users/dnowacki/compliance-checker/catalog.html?dataset=sand/usgs/users/dnowacki/compliance-checker/DL115LWAvir.nc).

**Comment by daltonkell** [comment]
Thanks for raising, @dnowacki-usgs, we'll begin looking at this shortly.

**Comment by daltonkell** [comment]
Successfully reproduced on `master` branch.

**Comment by daltonkell** [comment]
Actually @dnowacki-usgs, I'm not getting the warning you seem to be getting. Can you confirm which version of the Checker you're using?

**Comment by Bobfrat** [comment]
> Actually @dnowacki-usgs, I'm not getting the warning you seem to be getting. Can you confirm which version of the Checker you're using?

I think the warning is secondary to the issue of the checker returning differing results. Could it be that its actually running CF1.7 instead of CF1.6 in that case?

**Comment by daltonkell** [comment]
I actually just ran two consecutive runs for `['acdd', 'cf:1.6']` and it produced different results:

__Run 1__
```
(env38) dalton@ubuntu:~/compliance-checker$ python issue_826.py                                                                    
Using cached standard name table v66 from /home/dalton/.local/share/compliance-checker/cf-standard-name-table-test-66.xml


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 5 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "SED_981" missing the following attributes:
* coverage_content_type

variable "Trb_980" missing the following attributes:
* coverage_content_type

variable "z" missing the following attributes:
* long_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_email not present
* geospatial_bounds not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* license not present
* processing_level not present
* publisher_name not present
* publisher_email not present


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 1 potential issue


                                     Errors                                     
--------------------------------------------------------------------------------
§9.1 Features and feature types
* Unidentifiable feature for variable longitude
* Unidentifiable feature for variable latitude
* Unidentifiable feature for variable z
```

__Run 2, changed nothing__
```
(env38) dalton@ubuntu:~/compliance-checker$ python issue_826.py 
Using cached standard name table v66 from /home/dalton/.local/share/compliance-checker/cf-standard-name-table-test-66.xml


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
All tests passed!


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 5 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "SED_981" missing the following attributes:
* coverage_content_type

variable "Trb_980" missing the following attributes:
* coverage_content_type

variable "z" missing the following attributes:
* long_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_email not present
* geospatial_bounds not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* license not present
* processing_level not present
* publisher_name not present
* publisher_email not present
```

Notice how the `cf:1.6` test is printing first on the second run. This indicates it might be an `lru_cache` problem.

To answer you question above, @Bobfrat, no, it's not running CF-1.7 instead of 1.6; 1.6 is run both times.

**Comment by dnowacki-usgs** [comment]
@daltonkell I only get the 
```
WARNING: The following exceptions occured during the cf:1.6 checker (possibly indicate compliance checker issues):
cf:1.6.check_data_types: type object 'str' has no attribute 'kind'
```
on my macOS machine. On a Linux machine, I don't get that warning. On both, `compliance_checker.__version__` returns `4.3.3`, and I installed both using conda-forge. In any case, this warning does seem to be secondary to the issue of the checker returning different results on subsequent runs.

**Comment by daltonkell** [comment]
To my last comment above: I believe the `lru_cache` problem is confirmed by the following steps:

1. Remove any `__pycache__/` directories
2. Run CF
3. Remove any `__pycache__/` directories
4. Run ACDD, then CF

My output is this:

```
(env38) dalton@ubuntu:~/compliance-checker$ rm -rf compliance_checker/__pycache__/ && rm -rf compliance_checker/cf/__pycache__/
(env38) dalton@ubuntu:~/compliance-checker$ vim issue_826.py # ensure only CF being run
(env38) dalton@ubuntu:~/compliance-checker$ python issue_826.py 
Using cached standard name table v66 from /home/dalton/.local/share/compliance-checker/cf-standard-name-table-test-66.xml


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
All tests passed!
(env38) dalton@ubuntu:~/compliance-checker$ 
(env38) dalton@ubuntu:~/compliance-checker$ 
(env38) dalton@ubuntu:~/compliance-checker$ 
(env38) dalton@ubuntu:~/compliance-checker$ 
(env38) dalton@ubuntu:~/compliance-checker$ 
(env38) dalton@ubuntu:~/compliance-checker$ rm -rf compliance_checker/__pycache__/ && rm -rf compliance_checker/cf/__pycache__/                                                        
(env38) dalton@ubuntu:~/compliance-checker$ python issue_826.py 
Using cached standard name table v66 from /home/dalton/.local/share/compliance-checker/cf-standard-name-table-test-66.xml


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
All tests passed!
(env38) dalton@ubuntu:~/compliance-checker$ rm -rf compliance_checker/__pycache__/ && rm -rf compliance_checker/cf/__pycache__/
(env38) dalton@ubuntu:~/compliance-checker$ vim issue_826.py  # ensure ACDD then CF                                                                                                                         
(env38) dalton@ubuntu:~/compliance-checker$ python issue_826.py 
Using cached standard name table v66 from /home/dalton/.local/share/compliance-checker/cf-standard-name-table-test-66.xml


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                     cf:1.6                                     
       http://cfconventions.org/cf-conventions/v1.6.0/cf-conventions.html       
--------------------------------------------------------------------------------
All tests passed!


--------------------------------------------------------------------------------
                         IOOS Compliance Checker Report                         
                                    acdd:1.3                                    
http://wiki.esipfed.org/index.php?title=Category:Attribute_Conventions_Dataset_Discovery
--------------------------------------------------------------------------------
                               Corrective Actions                               
DL115LWAvir.nc has 5 potential issues


                               Highly Recommended                               
--------------------------------------------------------------------------------
Global Attributes
* keywords not present
* summary not present

variable "SED_981" missing the following attributes:
* coverage_content_type

variable "Trb_980" missing the following attributes:
* coverage_content_type

variable "z" missing the following attributes:
* long_name


                                  Recommended                                   
--------------------------------------------------------------------------------
Global Attributes
* acknowledgment/acknowledgement not present
* comment not present
* creator_name not present
* creator_email not present
* geospatial_bounds not present
* geospatial_vertical_min not present
* geospatial_vertical_max not present
* geospatial_vertical_positive not present
* geospatial_bounds_crs not present
* geospatial_bounds_vertical_crs not present
* license not present
* processing_level not present
* publisher_name not present
* publisher_email not present
```

This seems _expected_ that results shouldn't change, but obviously having to remove the cache is an issue.

**Comment by neumannd** [comment]
@dnowacki-usgs the warning `type object 'str' has no attribute 'kind'` is a known issue (#809 )

**Comment by daltonkell** [comment]
Closing as this is resolved with #827
